src/modules/esl-event-listener/core/targets/wheel.target.ts

```diff
@@ -63,12 +70,15 @@ export class ESLWheelTarget extends SyntheticEventTarget {
   /** Handles wheel events */
   @bind
   protected _onWheel(event: WheelEvent): void {
+    this.scrollData = this.scrollData.concat(getParentScrollOffsets(event.target as Element, this.target));
     this.aggregateWheel(event);
   }
 
   /** Handles aggregated wheel events */
   protected handleAggregatedWheel(events: WheelEvent[]): void {
     const wheelInfo =  this.resolveEventDetails(events);
+
+    if (this.config.processOverflow && isOffsetChanged(this.scrollData)) return;
```

Comment:
```suggestion
    const isBlocked = isOffsetChanged(this.scrollData);
    this.scrollData = [];
    if (isBlocked) return;
```

1. Critical: we need to clear scroll data, otherwise it is a leak.
2. `this.config.processOverflow` could be committed here in case we take it into account in `scrollData` management. (`[].some(...)` is an elementary operation, so we can skip additional checks in that case)